Rescue and transform sequel database errors in Lotus::Command

chongfun opened the pull request:
When creating an entity, the SqlAdapter does not raise a Lotus::Model::Error and
instead bubbles exceptions up from the adapter.  Applications should not be
concerned with the underlying adapter implementation, so the sql adapter command
has been modified to transform sequel database errors into lotus errors.

chongfun committed:
Rescue and transform sequel database errors in Lotus::Command

When creating an entity, the SqlAdapter does not raise a Lotus::Model::Error and
instead bubbles exceptions up from the adapter.  Applications should not be
concerned with the underlying adapter implementation, so the sql adapter command
has been modified to transform sequel database errors into lotus errors.

drewdeponte commented on lib/lotus/model.rb:
```diff
@@ -38,6 +38,34 @@ def initialize(message = "Invalid command")
       end
     end
 
+    # Error raised when a check constraint has been violated
```
I would rename this comment to "Error for check constraint violation". I would do this so that the format of the documentation and style better matches the existing pseudo standard.

chongfun commented:
@BRIMIL01 Do you have any suggestions for testing that `#create`, `#update`, and `#delete` always rescue `Sequel::DatabaseError`?

chongfun committed:
Transform Sequel::DatabaseError into Lotus::Model::Error

This transformation is sufficient because Exception#cause makes the original
exception available.

chongfun commented:
Closing since I've opened a PR against lotus/model.
